flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/connectors/CollectDynamicSink.java

```diff
@@ -93,39 +98,40 @@ public ChangelogMode getChangelogMode(ChangelogMode requestedMode) {
 
     @Override
     public SinkRuntimeProvider getSinkRuntimeProvider(Context context) {
-        return (DataStreamSinkProvider)
-                inputStream -> {
```

Comment:
did we break the API for lambdas now?

Reply:
yes :cry: the only way to not break it is to not default implement the new method